.github/actions/diffs/action.yml

```diff
@@ -0,0 +1,26 @@
+name: Detect Changes
+description: Defines variables indicating the parts of the code that changed
+outputs:
+  isRust:
+    description: True when changes happened to the rust code
+    value: "${{ steps.diff.outputs.isRust }}"
+  isExplorerClient:
+    description: True when there are changes to files related to explorer client
+    value: "${{ steps.diff.outputs.isExplorerClient }}"
+runs:
+  using: composite
+  steps:
+  - uses: actions/checkout@v3
+  - name: Detect Changes
+    uses: dorny/paths-filter@v2.10.2
+    id: diff
+    with:
+      filters: |
+        isRust:
+          - '!(explorer|doc|.github|sdk)/**'
```

Comment:
One thing that I'd like to implement is changing up the directory hierarchy such that all rust crates must live in a `/crates` directory, it would make things a bit easier from the perspective of figuring out where/when rust code changes at the very least.

Reply:
Agree! Added https://github.com/MystenLabs/sui/issues/1507 - feel free to refine

I was thinking also, since we have multiple projects in the repo (and probably will add more), that we should have a more generic workflow/pipeline which can detect changes and run project specific tasks/jobs that are defined under the project directory. (Probably quite complicated and maybe not worth the effort for now though - easier to use multiple workflow file for each project)